common/djangoapps/third_party_auth/tasks.py

```diff
@@ -36,16 +37,13 @@ def fetch_saml_metadata():
         num_failed: Number of providers that could not be updated
         num_total: Total number of providers whose metadata was fetched
     """
-    if not SAMLConfiguration.is_enabled():
-        return (0, 0, 0)  # Nothing to do until SAML is enabled.
-
     num_changed, num_failed = 0, 0
 
     # First make a list of all the metadata XML URLs:
     url_map = {}
     for idp_slug in SAMLProviderConfig.key_values('idp_slug', flat=True):
         config = SAMLProviderConfig.current(idp_slug)
-        if not config.enabled:
+        if not config.enabled or SAMLConfiguration.is_enabled(config.site):
```

Comment:
Should this be `if not config.enabled or not SAMLConfiguration.is_enabled(config.site):`?